mercury-framework/mercury-grpc-service/src/main/java/com/redhat/mercury/GrpcService.java

```diff
@@ -0,0 +1,123 @@
+package com.redhat.mercury;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import com.google.inject.Inject;
+import com.redhat.mercury.model.FuntionalPattern;
+import com.redhat.mercury.resources.StringFormatter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * This service generates a gRPC client connection for all APIs available in the service domain
+ */
+@Mojo(name = "GrpcService")
+public class GrpcService extends AbstractMojo {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GrpcService.class);
+
+    @Inject
+    StringFormatter formatter;
+
+    @Parameter(property = "serviceDomainName", required = true)
+    private String sdName;
+
+    @Parameter(property = "version", required = false, defaultValue = "v10")
+    private String version;
+
+    @Parameter(property = "port", required = false, defaultValue = "9000")
+    private String port;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        String sdNameCamel = formatter.capitalizeWords(sdName.replace("-", " "));
+
+        // Retrieve list of apis from the proto/api directory
+        String[] apiFiles = new File(String.format("service-domains/%s/%s-common/src/main/proto/%s/api",
```

Comment:
this path should be relative to the project as this plugin is intended to be used in the scope of a service domain project